Source/SwiftLintFramework/Extensions/Configuration+LintableFiles.swift

```diff
@@ -13,14 +23,40 @@ extension Configuration {
             return [path]
         }
         let pathsForPath = included.isEmpty ? fileManager.filesToLint(inPath: path, rootDirectory: nil) : []
-        let excludedPaths = excluded.flatMap {
-            fileManager.filesToLint(inPath: $0, rootDirectory: rootPath)
-        }
+        let excludedPaths = excluded
+            .flatMap { resolveGlobs(in: $0) }
+            .flatMap {
+                fileManager.filesToLint(inPath: $0, rootDirectory: rootPath)
+            }
         let includedPaths = included.flatMap {
             fileManager.filesToLint(inPath: $0, rootDirectory: rootPath)
         }
         return (pathsForPath + includedPaths).filter {
             !excludedPaths.contains($0)
         }
     }
+
+    private func resolveGlobs(in pattern: String) -> [String] {
+        guard pattern.contains("*") else {
+            return [pattern]
+        }
+
+        var globResult = glob_t()
+        defer { globfree(&globResult) }
+
+        let flags = GLOB_TILDE | GLOB_BRACE | GLOB_MARK
+        guard globFunction(pattern.cString(using: .utf8)!, flags, nil, &globResult) == 0 else {
+            return []
+        }
+
+#if os(Linux)
+        let matchCount = globResult.gl_pathc
+#else
+        let matchCount = globResult.gl_matchc
+#endif
+
+        return (0..<Int(matchCount)).compactMap { index in
+            return String(validatingUTF8: globResult.gl_pathv[index]!)
```

Comment:
what do you think about `return globResult.gl_pathv[index].flatMap(String.init(validatingUTF8:))` just to avoid any potencial issues?

Reply:
Done, but FWIW in this case IMO it's better to crash since this would be breaking the API contract.

Reply:
Had to use it in a closure since it couldn't infer the type otherwise